services/github/github-commit-activity.service.js

```diff
@@ -46,47 +52,67 @@ export default class GithubCommitActivity extends GithubAuthV3Service {
     }
   }
 
-  static transform({ interval, weekData }) {
-    const weekTotals = weekData.map(({ total }) => total)
+  async fetch({ interval, user, repo, branch = 'FETCH_HEAD' }) {
+    const since = this.constructor.getIntervalQueryStartDate({ interval })
+    return this._requestGraphql({
+      query: gql`
+        query (
+          $user: String!
+          $repo: String!
+          $branch: String!
+          $since: GitTimestamp!
+        ) {
+          repository(owner: $user, name: $repo) {
+            object(expression: $branch) {
+              ... on Commit {
+                history(since: $since) {
+                  totalCount
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        user,
+        repo,
+        branch,
+        since,
+      },
+      schema,
+      transformErrors,
+    })
+  }
+
+  static transform({ data }) {
+    const {
+      repository: { object: repo },
+    } = data
 
-    if (interval === 'm') {
-      // To approximate the value for the past month, get the sum for the last
-      // four weeks and add a weighted value for the fifth week.
-      const fourWeeksValue = weekTotals
-        .slice(-4)
-        .reduce((sum, weekTotal) => sum + weekTotal, 0)
-      const fifthWeekValue = weekTotals.slice(-5)[0]
-      const averageWeeksPerMonth = 365 / 12 / 7
-      return (
-        fourWeeksValue + Math.round((averageWeeksPerMonth - 4) * fifthWeekValue)
-      )
+    if (!repo) {
+      throw new InvalidResponse({ prettyMessage: 'invalid branch' })
     }
 
-    let wantedWeekData
-    switch (interval) {
-      case 'y':
-        wantedWeekData = weekTotals
-        break
-      case '4w':
-        wantedWeekData = weekTotals.slice(-4)
-        break
-      case 'w':
-        wantedWeekData = weekTotals.slice(-2, -1)
-        break
-      default:
-        throw Error('Unhandled case')
+    return repo.history.totalCount
+  }
+
+  static getIntervalQueryStartDate({ interval }) {
+    const now = new Date()
+
+    if (interval === 'y') {
+      now.setUTCFullYear(now.getUTCFullYear() - 1)
+    } else if (interval === 'm' || interval === '4w') {
+      now.setUTCDate(now.getUTCDate() - 30)
```

Comment:
The classic fun question of "what is a month, really?" :laughing: I know we still have momentjs in our dep tree but I opted to avoid using it to prevent from increasing our dependence on it due to #5573.

IMO showing the commit activity for the last 30 days is a reasonable time period to cover both of these intervals, but suppose we could try to do something more complex. I feel like some of the complexity we had before was more due to the nature of the v3 API response that just gave us total counts in weekly slices, but lmk if folks have any concerns with treating month as the last 30 days

Reply:
There are other badges where we are making the same assumption e.g:
https://github.com/badges/shields/blob/b02471ff425f1db13d4786f1c7ca933d868f8242/services/packagecontrol/packagecontrol.service.js#L56-L71
https://github.com/badges/shields/blob/b02471ff425f1db13d4786f1c7ca933d868f8242/services/sourceforge/sourceforge.service.js#L22-L24
https://github.com/badges/shields/blob/b02471ff425f1db13d4786f1c7ca933d868f8242/services/homebrew/homebrew-downloads.service.js#L19-L23
https://github.com/badges/shields/blob/b02471ff425f1db13d4786f1c7ca933d868f8242/services/wordpress/wordpress-downloads.service.js#L31-L34
I don't know where it fits in the docs, but I'd be happy to document somewhere that if the upstream api doesn't explicitly provide stats for "month", we assume month = 30 days and formalise this.

Reply:
Cool, glad to be in good company!